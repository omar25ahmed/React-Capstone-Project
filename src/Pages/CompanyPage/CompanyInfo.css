ul {
  padding: 0;
  margin: 0;
  list-style: none;
}

li {
  height: 50px;
  padding: 15px;
}
