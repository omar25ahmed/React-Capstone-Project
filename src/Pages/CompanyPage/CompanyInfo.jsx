import React from 'react';
import PropTypes from 'prop-types';
import './CompanyInfo.css';

function CompanyInfo({ company }) {
  const {
    marketCap, sector, industry, price, exchange, exchangeShortName, isActivelyTrading,
  } = company;
  const nf = new Intl.NumberFormat('en-US');
  return (
    <ul data-testid="company info">
      <li className="d-flex justify-content-between">
        <span>Market Cap</span>
        {nf.format(marketCap)}
        {' '}
        $
      </li>
      <li className="d-flex justify-content-between">
        <span>Sector</span>
        {sector}
      </li>
      <li className="d-flex justify-content-between">
        <span>Industry</span>
        {industry}
      </li>
      <li className="d-flex justify-content-between">
        <span>Price of the company</span>
        {price}
        {' '}
        M
      </li>
      <li className="d-flex justify-content-between">
        <span>Exchange</span>
        {exchange}
      </li>
      <li className="d-flex justify-content-between">
        <span>Exchange shortcut</span>
        {exchangeShortName}
      </li>
      <li className="d-flex justify-content-between">
        <span>Trading status</span>
        {isActivelyTrading ? 'active' : 'not active'}
      </li>
    </ul>
  );
}

export default CompanyInfo;

CompanyInfo.defaultProps = {
  company: {
    companyName: '',
    symbol: '',
  },
};

CompanyInfo.propTypes = {
  company: PropTypes.shape({
    marketCap: PropTypes.number,
    sector: PropTypes.string,
    industry: PropTypes.string,
    price: PropTypes.number,
    exchange: PropTypes.string,
    exchangeShortName: PropTypes.string,
    isActivelyTrading: PropTypes.bool,
  }),
};
