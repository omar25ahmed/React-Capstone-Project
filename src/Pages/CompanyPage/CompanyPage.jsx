import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import CompanyInfo from './CompanyInfo';
import { fetchCompanies } from '../../Redux/companiesReducer';
import HeadingComponents from '../HeadingComponent';

function CompanyPage() {
  const { id } = useParams();
  const dispatch = useDispatch();
  const companies = useSelector((state) => state);
  useEffect(() => {
    dispatch(fetchCompanies());
  }, []);
  return (
    <div data-testid="company" className="container">
      {companies
        .filter((company) => company.companyName === id && company)
        .map((company) => (
          <div key={Math.random() * 100}>
            <HeadingComponents
              key={Math.random()}
              name={company.companyName}
              title={company.symbol}
            />
            <CompanyInfo company={company} />
          </div>
        ))}
    </div>
  );
}

export default CompanyPage;
