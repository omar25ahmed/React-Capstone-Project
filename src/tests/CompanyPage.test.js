import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Redux/Store';
import CompanyPage from '../Pages/CompanyPage/CompanyPage';

describe('testing Home Company Page', () => {
  it('snapshot for the Company Page ', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CompanyPage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing CompanyPage component renders correctly', () => {
    render(
      <Provider store={store}>
        <CompanyPage />
      </Provider>,
    );
    expect(screen.getByTestId('company')).toBeInTheDocument();
  });
});
