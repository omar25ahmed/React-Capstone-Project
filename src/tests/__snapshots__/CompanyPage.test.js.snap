// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home Company Page snapshot for the Company Page  1`] = `
<div
  className="container"
  data-testid="company"
/>
`;
